### ISSUE TITLE
Material list using decimals

### ISSUE THREAD

**Issue by LiquidFenrir** [MauveCloud/Ic2ExpReactorPlanner#24]
The design I had in the grid had 7 reactor platings (each costing 1 lead plate, and 1 advanced alloy, so a total of 7 advanced alloy), but the material list said "7,000 * advanced alloy" needed.

Not that bad once you notice it, nor hard to fix by head (just remove the zeros), but it did make my heart jump, I thought it meant 7 thousand.

You shouldnt really use that in my opinion. if there's leftover, just add 1 to the total (for example, the RSH-Condensator costing "12,333 * Copper" should be "13 * Copper" as that is what it will really cost.

**Comment by MauveCloud** [comment]
I asked two years ago in the thread about the planner whether to give exact amounts or round up, and Blackpalt said I should give exact amounts (see https://forum.industrial-craft.net/thread/10998-ic2-experimental-reactor-planner-open-beta/?postID=192115#post192115 ).  As far as "what it will really cost", if you're making multiple copies of a reactor design, that's no longer the case.  If you want to join the thread there, and enough others agree that rounding up is better, I'll change it.

As far as thinking it was 7 thousand, I'm guessing that has something to do with your regional settings - some countries use a comma as a decimal separator and a period as a thousands grouper (opposite the way it is done in the United States of America), and the planner picks that up via the Java api instead of forcing US style on everyone.  However, somehow I thought I had already added handling to skip showing the decimal part when the number is close enough to an exact integer.  Maybe I'm misremembering, though, so I'll look into that when I have a chance.

**Comment by LiquidFenrir** [comment]
screenshot of the materials list: https://i.imgur.com/rHPWbwX.png
I'm french, we use `,` as the decimals separator and `.` or nothing as thousands grouper
And it's alright, was just my opinion on that

**Comment by MauveCloud** [comment]
I'm confused.  If you're accustomed to using a comma as the decimal separator, then why did you briefly think it meant 7 thousand?

Anyway, I had added such handling (here: https://github.com/MauveCloud/Ic2ExpReactorPlanner/commit/3524c6c0353d6ad33c71e354e77e6c132ce723ab#diff-7a35237c2355e75a0ca1c2a7d7601287 ), only that fix doesn't work properly with different decimal separators.

**Comment by LiquidFenrir** [comment]
using `count % 1 == 0` instead of `formattedNumber.endsWith(".000")` should fix it, and be faster

**Comment by MauveCloud** [comment]
> using count % 1 == 0 instead of formattedNumber.endsWith(".000") should fix it, and be faster

Actually, no.  The Java language specification is a little confusing (at least to me) when it comes to how the remainder operator works on floating-point numbers, but I tried a short loop using that, and it's not a reliable test for the number being "close enough" to an integer value (which is what's really needed here):
```
x = 5.9997; x % 1 = 0.9996999999999998
x = 5.9998; x % 1 = 0.9997999999999996
x = 5.999899999999999; x % 1 = 0.9998999999999993
x = 5.999999999999999; x % 1 = 0.9999999999999991
x = 6.000099999999999; x % 1 = 9.999999999887876E-5
x = 6.000199999999999; x % 1 = 1.999999999986457E-4
x = 6.000299999999998; x % 1 = 2.9999999999841265E-4
```

**Comment by MauveCloud** [comment]
Fixed in v2.1.6.
